SqlPersistence/Timeout/SequentialGuid.cs

```diff
@@ -0,0 +1,33 @@
+namespace NServiceBus.Persistence.Sql
+{
+    using System;
+    using System.Threading;
+
+    // From https://github.com/aspnet/EntityFramework/blob/dev/src/Microsoft.EntityFrameworkCore/ValueGeneration/SequentialGuidValueGenerator.cs
+    static class SequentialGuid
+    {
+        static long counter = DateTime.UtcNow.Ticks;
```

Comment:
Is this algorithm safe? I guess the change that of conflict is small when using ticks but maybe we should include some machine identification also (MAC?). Like they propose here: http://stackoverflow.com/questions/1752004/sequential-guid-generator